src/services/analytics/useGtm.ts

```diff
@@ -79,6 +82,10 @@ const useGtm = () => {
     gtmTrackPageview(router.pathname)
   }, [router.pathname])
 
+  useEffect(() => {
+    gtmSetUserProperty(AnalyticsUserProperties.WALLET_LABEL, walletLabel ?? WALLET_LABEL_NONE)
+  }, [walletLabel])
```

Comment:
Since the wallet is asynchronous, it will inevitably log NONE for the first few events before the wallet (re)connects. I would suggest to not set it at all until the user connects a wallet.

Optionally, we could also unset it when the user disconnects. You migth have to listen to a specific onboard event for that. But to keep things simple I would just keep the last wallet and never unset it.

Reply:
Makes sense. Especially because the interesting interactions such as signing a message / executing a tx will require a wallet anyways.